src/posix/platform/hdlc_interface.cpp

```diff
@@ -656,6 +667,88 @@ void HdlcInterface::HandleHdlcFrame(otError aError)
     }
 }
 
+#if OPENTHREAD_SPINEL_CONFIG_RESET_CONNECTION
+otError HdlcInterface::ResetConnection(void)
+{
+    otError error = OT_ERROR_NONE;
+
+    SuccessOrExit(error = WaitForUsbDevice(mRadioUrl.GetPath()));
+
+    CloseFile();
+
+    mSockFd = OpenFile(mRadioUrl);
+    VerifyOrExit(mSockFd != -1, error = OT_ERROR_FAILED);
+
+exit:
+    return error;
+}
+
+otError HdlcInterface::WaitForUsbDevice(const char *aRadioUrlPath)
+{
+    int                  fd;
+    uint64_t             end;
+    uint64_t             now;
+    struct udev_monitor *mon;
+    otError              error = OT_ERROR_NONE;
+
+    struct udev *udev = udev_new();
+    VerifyOrExit(udev != nullptr, error = OT_ERROR_FAILED);
+
+    mon = udev_monitor_new_from_netlink(udev, "udev");
+    udev_monitor_filter_add_match_subsystem_devtype(mon, "tty", NULL);
+    udev_monitor_enable_receiving(mon);
+    fd = udev_monitor_get_fd(mon);
+
+    // wait maximally 10 seconds
+    end = otPlatTimeGet() + 10 * US_PER_S;
+    do
+    {
+        int            ret;
+        fd_set         fds;
+        struct timeval tv;
+
+        tv.tv_sec  = 0;
+        tv.tv_usec = 100 * US_PER_MS;
+
+        FD_ZERO(&fds);
+        FD_SET(fd, &fds);
+
+        ret = select(fd + 1, &fds, NULL, NULL, &tv);
+        if (ret > 0 && FD_ISSET(fd, &fds))
+        {
+            struct udev_device *dev = udev_monitor_receive_device(mon);
+            if (dev)
+            {
+                struct udev_list_entry *entry;
+                const char *            action = udev_device_get_action(dev);
+
+                if (strncmp(action, "add", strlen("add")) == 0)
+                {
+                    udev_list_entry_foreach(entry, udev_device_get_devlinks_list_entry(dev))
+                    {
+                        const char *name = udev_list_entry_get_name(entry);
+                        if (strncmp(name, aRadioUrlPath, strlen(aRadioUrlPath)) == 0)
```

Comment:
```suggestion
                        if (strcmp(name, aRadioUrlPath) == 0)
```

The original code would think that `/dev/ttyACM11`  equals to `/dev/ttyACM1` ?